src/components/MultiHopTrade/hooks/useGetTradeQuotes.tsx

```diff
@@ -0,0 +1,77 @@
+import { getTradeQuoteArgs } from 'components/Trade/hooks/useSwapper/getTradeQuoteArgs'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import type { GetTradeQuoteInput } from 'lib/swapper/api'
+import { SwapperName } from 'lib/swapper/api'
+import {
+  selectBuyAsset,
+  selectPortfolioAccountIdsByAssetId,
+  selectPortfolioAccountMetadataByAccountId,
+  selectReceiveAddress,
+  selectSellAmountCryptoPrecision,
+  selectSellAsset,
+  selectSellAssetAccountId,
+} from 'state/slices/selectors'
+import { useGetLifiTradeQuoteQuery } from 'state/slices/swappersSlice/swappersSlice'
+import { useAppSelector } from 'state/store'
+
+export const useGetTradeQuotes = () => {
+  const isLifiEnabled = useFeatureFlag('LifiSwap')
+  const wallet = useWallet().state.wallet
+  const [tradeQuoteInput, setTradeQuoteInput] = useState<GetTradeQuoteInput | undefined>()
+  const sellAsset = useAppSelector(selectSellAsset)
+  const buyAsset = useAppSelector(selectBuyAsset)
+  const sellAssetAccountId = useAppSelector(selectSellAssetAccountId)
+  const receiveAddress = useAppSelector(selectReceiveAddress)
+  const sellAmountCryptoPrecision = useAppSelector(selectSellAmountCryptoPrecision)
+
+  const sellAssetAccountIds = useAppSelector(state =>
+    selectPortfolioAccountIdsByAssetId(state, {
+      assetId: sellAsset.assetId,
+    }),
+  )
+  const sellAccountFilter = { accountId: sellAssetAccountId ?? sellAssetAccountIds[0] }
+  const sellAccountMetadata = useAppSelector(state =>
+    selectPortfolioAccountMetadataByAccountId(state, sellAccountFilter),
+  )
+
+  useEffect(() => {
+    if (wallet && sellAccountMetadata) {
+      ;(async () => {
+        const { chainId: receiveAddressChainId } = fromAssetId(buyAsset.assetId)
+        const chainAdapter = getChainAdapterManager().get(receiveAddressChainId)
+
+        if (!chainAdapter)
+          throw new Error(`couldn't get chain adapter for ${receiveAddressChainId}`)
+        const { accountNumber: sellAccountNumber } = sellAccountMetadata.bip44Params
+
+        const tradeQuoteInputArgs: GetTradeQuoteInput | undefined = await getTradeQuoteArgs({
+          sellAsset,
+          sellAccountNumber,
+          sellAccountType: sellAccountMetadata.accountType,
+          buyAsset,
+          wallet,
+          receiveAddress,
+          sellAmountBeforeFeesCryptoPrecision: sellAmountCryptoPrecision,
+        })
+        setTradeQuoteInput(tradeQuoteInputArgs)
+      })()
+    } else {
+      setTradeQuoteInput(undefined)
+    }
+  }, [buyAsset, receiveAddress, sellAccountMetadata, sellAmountCryptoPrecision, sellAsset, wallet])
+
+  const { isLoading, data, error } = useGetLifiTradeQuoteQuery(tradeQuoteInput ?? skipToken, {
+    skip: !isLifiEnabled,
+  })
+
+  if (!tradeQuoteInput) return {}
+
+  return {
+    [SwapperName.LIFI]: { isLoading, data, error },
```

Comment:
I think it makes sense because the return type is an unordered set of quotes, and the empty set is `{}` in this case. 

If we chose to make the empty set `undefined` it would required loads of `?.` downstream which only makes sense if we needed to differentiate between "no quotes were found" and "there wasn't a valid trade input". Input validation is handled upstream to this hook and consumers of this only care about quote data rather than input validation.

Reply:
alternatively i could make it return `{ [SwapperName.LIFI]: undefined }` but i dont see an advantage to this

Reply:
c8a99c64e6d276d8ec85104d363d4f8e62fd0a66

Reply:
actually is totally reworked in a downstream branch so no use refactoring now :)